spotbugs/src/main/java/edu/umd/cs/findbugs/BugInstance.java

```diff
@@ -2113,11 +2113,11 @@ private void addJavaAnnotationNames(BugAnnotation annotation) {
             ClassContext classContext = analysisCache.getClassAnalysis(ClassContext.class, classDescriptor);
             JavaClass javaClass = classContext.getJavaClass();
             AnnotationEntry[] annotationEntries = javaClass.getAnnotationEntries();
-            List<String> javaAnnotationNames = Arrays.asList(annotationEntries).stream().map((AnnotationEntry ae) -> {
-                // map annotation entry type to dotted class name, for example
-                // Lorg/immutables/value/Generated; -> org.immutables.value.Generated
-                return ae.getAnnotationType().substring(1).replace("/", ".").replace(";", "");
-            }).collect(Collectors.toList());
+            // map annotation entry type to dotted class name, for example
+            // Lorg/immutables/value/Generated; -> org.immutables.value.Generated
+            List<String> javaAnnotationNames = Arrays.asList(annotationEntries).stream()
+                    .map((AnnotationEntry ae) -> ClassName.fromFieldSignatureToDottedClassName(ae.getAnnotationType()))
+                    .collect(Collectors.toList());
```

Comment:
This is the only change where it's not only the usage of the method from the ClassName util class, but also removing the unnecessary block and return, and moving the comment.